### PR TITLE
feat: adds user authentication

### LINKED ISSUE #40: Move logic of navbar from presentational component to container component
Logic for the popup is in the presentational component navbar should be moved to a container component for navbar

### PR DESCRIPTION
**Pull request by blendaf**
fixes: #40

### CODE REVIEW

File: container/FireBaseAuthProvider.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const firebaseApp = !firebase.apps.length
+  ? firebase.initializeApp(config)
+  : firebase.app()
+const firebaseAppAuth = firebaseApp.auth()
+
+const providers = {
+  googleProvider: new firebase.auth.GoogleAuthProvider()
+}
+
+export const MyContext = React.createContext()
```
Comment (juliuscc) [inline container/FireBaseAuthProvider.js]: Could this possibly have a name that better explains what it is? Maybe AuthContext or UserContext

**Comment by Falcasantos** [comment]
does the conflict in package.json mean that "awesome-only-resolves-last-promise" has been deleted? Do we need this?

**Comment by juliuscc** [comment]
> does the conflict in package.json mean that "awesome-only-resolves-last-promise" has been deleted? Do we need this?

Probably, and yes we need it. We need it in the sense that the build would fail, not that we can't implement it ourself.

It does what the teacher calls "debouncing". We could implement it ourself but I struggled with doing that.

**Comment by juliuscc** [comment]
The conflict is fixed

File: container/FireBaseAuthProvider.js (hunk 1 above)
Comment (blendaf) [reply container/FireBaseAuthProvider.js]: Will fix as soon as Catarina is done with the branch

**Review by juliuscc — APPROVED** [review]
LGTM!

Ship it!

gif of ship

**Pull request merged**